### DISCUSSION TITLE
How to reset a server-side fetch when it becomes outdated

### DISCUSSION THREAD

**Discussion by lextoc** [vercel/next.js#53050] [Help] [answered]
### Summary

I'm wondering how I can recall the validateToken fetch if somewhere down below (nested in Navigation for example) a user signs out. What is a good way to clear the user variable (or set a user to it when the cookies become available)? I'm trying to render the user server-side so that's why I cannot use hooks or similar.

Layout.tsx:
```typescript
import { cookies } from "next/headers";

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  const response = await validateToken(cookies());
  let user = null;
  if (response.success) user = response.data;

  return (
    <html lang="en">
      <body>
        <Navigation user={user}
        {children}
      </body>
    </html>
  );
}
```

When I refresh the page the user data is always up to date so server-side it is working.

**Comment by thanhtutzaw** [comment]
You want to use user in SSR ? Then retrieve cookies value from Context in SSR . like this (with nookies)

```
export const getServerSideProps: GetServerSideProps<Props> = async (
  context
) => {
  try {
    const cookies = nookies.get(context);
    const token = (await verifyIdToken(cookies.token)) as DecodedIdToken;

```

And when userSignout or IdToken change (I use firebase auth)
 remove cookies

**Reply by lextoc** [reply]
I'm getting the following error: 
`"getServerSideProps" is not supported in app/.`

![PixelSnap 2023-07-23 at 09 38 12](https://github.com/vercel/next.js/assets/4405004/677b0ef9-67bd-4b5e-b37f-c8304ea2b463)

**Reply by lextoc** [reply]
I'm trying to use [the new approach](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching).

**Reply by lextoc** [reply]
Also, the cookies are actually retrieved server side. The SSR is working , I'm wondering how I can update the user if he/she signs out after the initial page load has been done. (Or signs in for that matter). With React Query you'd do something like `queryClient.invalidateQueries` but that's not possible because you'd need a client component and then SSR doesn't work anymore...

**Reply by lextoc** [reply]
Code is available here for anyone who wants to try to help me out. Any help is greatly appreciated. https://github.com/lextoc/tracky-next

**Comment by lextoc** [comment, marked as answer]
I've resolved the issue (was simpler than I thought 😅) with React Query's `initialData`. I've still got an issue where the data is not always displayed but that's going to be something client-side React Query related. For those interested here you can find the solution:
https://github.com/lextoc/tracky-next/blob/main/app/layout.tsx#L67

**Reply by lextoc** [reply]
Actually it's still not server side rendered but it improved the speed a lot.. Best as it gets I think
